src/main/java/com/metamx/common/parsers/JSONPathParser.java

```diff
@@ -60,7 +59,6 @@
    */
   public JSONPathParser(List<FieldSpec> fieldSpecs, boolean useFieldDiscovery, ObjectMapper mapper)
   {
-    this.fieldSpecs = fieldSpecs;
```

Comment:
@drcrallen fieldsSpecs are used in the next line, to derive some other field, while are not needed as a field itself.